src/java.base/windows/native/libnio/ch/WindowsAsynchronousServerSocketChannelImpl.c

```diff
@@ -131,7 +131,9 @@ Java_sun_nio_ch_WindowsAsynchronousServerSocketChannelImpl_updateAcceptContext(J
     SOCKET s1 = (SOCKET)jlong_to_ptr(listenSocket);
     SOCKET s2 = (SOCKET)jlong_to_ptr(acceptSocket);
 
-    setsockopt(s2, SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, (char *)&s1, sizeof(s1));
+    if (setsockopt(s2, SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, (char *)&s1, sizeof(s1)) < 0) {
```

Comment:
I think this goes back to when the JDK targeted Windows releases where this wasn't supported or there were non-MS LSPs installed. It should be okay to change now.

Reply:
Hi Alan, in the MS resources https://learn.microsoft.com/en-us/windows/win32/api/mswsock/nf-mswsock-acceptex  I find some info about 'Windows XP or later'  regarding the SO_UPDATE_ACCEPT_CONTEXT option. So maybe this was the Windows release you are referring to.